internal/host/hostcommands/instruction_manager.go

```diff
@@ -73,13 +73,14 @@ func NewInstructionManager(log logrus.FieldLogger, db *gorm.DB, hwValidator hard
 	ntpSynchronizerCmd := NewNtpSyncCmd(log, instructionConfig.AgentImage, db)
 	diskPerfCheckCmd := NewDiskPerfCheckCmd(log, instructionConfig.AgentImage, hwValidator, instructionConfig.DiskCheckTimeout.Seconds())
 	imageAvailabilityCmd := NewImageAvailabilityCmd(log, db, ocRelease, versionHandler, instructionConfig)
+	domainNameResolutionCmd := NewdomainNameResolutionCmd(log, instructionConfig.AgentImage, db)
 
 	return &InstructionManager{
 		log: log,
 		db:  db,
 		installingClusterStateToSteps: stateToStepsMap{
 			models.HostStatusKnown:                    {[]CommandGetter{connectivityCmd, freeAddressesCmd, dhcpAllocateCmd, inventoryCmd, ntpSynchronizerCmd}, defaultNextInstructionInSec},
-			models.HostStatusInsufficient:             {[]CommandGetter{inventoryCmd, connectivityCmd, freeAddressesCmd, dhcpAllocateCmd, ntpSynchronizerCmd}, defaultNextInstructionInSec},
+			models.HostStatusInsufficient:             {[]CommandGetter{inventoryCmd, connectivityCmd, freeAddressesCmd, dhcpAllocateCmd, ntpSynchronizerCmd, domainNameResolutionCmd}, defaultNextInstructionInSec},
```

Comment:
I think we'll want this in many other states, including day2 states (addHostsClusterToSteps). Not sure which exactly. Maybe @ori-amizur can say

Reply:
day2 seems logical in all pre-install states.
For day1, I am not sure if the requirement is that for every cluster, the api vip ip address is resolvable.  After all we either supply the VIP IPs, or we get them from DHCP.  In the last case, this will not work in insufficient status.
@ronniel1 ?

Reply:
@ori-amizur this validation is targeting user-managed-networking (none platform) installations.
There are no VIPs (that we know of)

Reply:
The validation should run only when UserManagedNetworking is set, so this condition was added to the validator, besides that, I've added the validation to all other states that similar to other network validation.